### DISCUSSION TITLE
When response_model is optional

### DISCUSSION THREAD

**Discussion by timconnorz** [instructor-ai/instructor-js#115] [General]
**response_model** is currently not an optional parameter, but it would be nice if it was for cases where its unnecessary.
In this case I expect Instructor would skip over the LLMValidator step.

**Comment by roodboi** [comment]
It should do a straight pass-through if the additional params are not provided - can you share under what circumstances this is not the case?

![CleanShot 2024-02-10 at 15 57 23@2x](https://github.com/instructor-ai/instructor-js/assets/344842/80c4889f-46cc-49bc-ba37-0d9584636bc2)

**Comment by timconnorz** [comment]
Ah my bad, I was confused by the errors I was getting. 

What confused me what the fact that when response_model is specified, you get your response in the exact shape of the model, but when it's omitted, you have to revert back to choices[0].delta and append them to the string yourself.

**Reply by roodboi** [reply]
Ahh got it - yeah the goal was to ensure full compatibility with the sdk, only adding the enhanced layer when needed. 

if all you want is a string response but don't need much else, could you still use the response model with just a simple schema?

eg


Z.object({ content: z.string() })

I almost never use straight sdk at this point, even if just for a single string response - for most cases I usually just want the string parsed out of the response and nothing else - this still gets you that and sets us up to add additional output should you need it.

**Reply by jxnl** [reply]
I Agree sometimes i use `Z.object({ content: z.string() })` since the type of response is often shorter and is less censored.
